src/main/java/Duke/Task/Deadline.java

```diff
@@ -0,0 +1,72 @@
+package Duke.Task;
+
+import Duke.Exceptions.DukeException;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Represents a task with a description and a due date.
+ */
+public class Deadline extends Task {
+    /**
+     * The due date of the task.
+     */
+    protected LocalDate by;
+
+    /**
+     * Constructs a new Deadline task with the given description and due date.
+     *
+     * @param description The description of the task.
+     * @param by The due date of the task in yyyy-MM-dd format.
+     * @throws DukeException If the date format is invalid.
+     */
+    public Deadline(String description, String by) throws DukeException {
+        super(description);
+
+        try {
+            this.by = LocalDate.parse(by);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Invalid date format! Please use yyyy-MM-dd!");
+        }
+    }
+
+    /**
+     * Constructs a new Deadline task with the given description and due date.
+     *
+     * @param description The description of the task.
+     * @param by The due date of the task in yyyy-MM-dd format.
+     * @param isDone Specifies if task is marked
+     * @throws DukeException If the date format is invalid.
+     */
+    public Deadline(String description, String by, boolean isDone) throws DukeException {
+        super(description, isDone);
+
+        try {
+            this.by = LocalDate.parse(by);
+        } catch (DateTimeParseException e) {
+            throw new DukeException("Invalid date format! Please use yyyy-MM-dd!");
+        }
+    }
+
+    /**
+     * Converts the task into a formatted string representation suitable for file storage on hard disk.
+     *
+     * @return A string representation of the task for file storage.
+     */
+    @Override
+    public String toFile() {
+        return "D | " + (isDone ? "1" : "0") + " | " + description + " | " + by;
```

Comment:
Is it better to separate this into multiple lines? It could improve code readability by using having each "+" term on a new line.

Alternatively, String.format could be useful too! 😄 

```suggestion
        return "D | "
            + (isDone ? "1" : "0")
            + " | "
            + description
            + " | "
            + by;
```

or even:

```suggestion
        return String.format("D | %d | %s | %s", isDone ? 1 : 0, description, by);
```